crates/polars-io/src/csv/read/read_impl/batched_read.rs

```diff
@@ -54,6 +55,8 @@ pub(crate) fn get_offsets(
     }
 }
 
+/// Reads bytes from `file` to `buf` and returns pointers into `buf` that can be parsed.
+/// TODO! this can be implemented without copying by pointing in the memmapped file.
```

Comment:
Currently this reads from `file` into `buf` and then returns offsets `(SyncPtr<u8>, usize)`, which is semantically `(*const u8, usize)`, being the starting offset ptr and the length of the chunk that can be parsed.

These pointers currently point into `buf`. However we can directly mmap the file and point into the memory mapped slice directly. This saves an explicit read and `reslice` (copy remaningin bytes to start of buf).